Enumerable.rb

```diff
@@ -1,114 +1,131 @@
+  # rubocop:disable Metrics/PerceivedComplexity
+  def my_any?(arg = nil, &block)
+    result = false
+    my_each do |i|
+      if arg.is_a?(Class) || arg.is_a?(Regexp)
+        result ||= i.is_a?(arg) unless arg.is_a?(Regexp)
+        result ||= !i.match(arg).nil? if arg.is_a?(Regexp)
+      elsif arg
+        result = true if i == arg
+      elsif block_given?
+        result ||= block.call(i) unless result
+      elsif i || result
         result = true
-        if block_given?
-            my_each { |i| result = false if yield(i) }
-        else
-            my_each { |i| result = false if i }
-        end 
-        result
+      end
     end
+    result
+  end
+  # rubocop:enable Metrics/CyclomaticComplexity
+  # rubocop:enable Metrics/PerceivedComplexity
 
-    def my_count(arg = nil)
-        result = 0
-        if block_given?
-            my_each do |i|
-                result += 1 if yield(i)
-            end
-        elsif arg
-            my_each do |i|
-                result += 1 if i == arg
-            end
-        else
-            result = length
-        end 
-        result
+  def my_none?(arg = nil, &block)
+    !my_any?(arg, &block)
+  end
+
+  def my_count(arg = nil)
+    result = 0
+    if block_given?
+      my_each do |i|
+        result += 1 if yield(i)
+      end
+    elsif arg
+      my_each do |i|
+        result += 1 if i == arg
+      end
+    else
+      result = length
     end
+    result
+  end
 
-    def my_map(proc = nil)
-        return to_enum unless block_given?
-        arr = []
-        if proc.nil?
-            length.times do |i|
-                arr.push(yield(self[i]))
-            end
-        else
-            length.times do |i|
-                arr.push(proc.call(self[i]))
-            end
-        end
-        arr
+  def my_map(proc = nil)
+    return to_enum unless block_given?
+
+    arr = []
+    if proc.nil?
+      length.times do |i|
+        arr.push(yield(self[i]))
+      end
+    else
+      length.times do |i|
+        arr.push(proc.call(self[i]))
+      end
     end
+    arr
+  end
 
-    def my_inject(arg1 = nil, arg2 = nil)
-        if block_given? && arg1
-            result = arg1
-            my_each do |i|
-                result = yield(result, i)
-            end 
-        elsif block_given? && !arg1
-            result = self[0]
-            my_each_with_index do |i, x|
-                result = yield(result, i) unless x.zero?
-            end
-        else
-            result = arg1
-            operation = arg2.to_proc
-            my_each do |i|
-                result = operation.call(result, i)
-            end
-        end 
-        result
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  def my_inject(arg = nil, arg2 = nil)
+    result = is_a?(Range) ? min : self[0]
+    if arg2.is_a?(Symbol) || arg2.is_a?(String)
+      my_each_with_index { |element, i| result = result.send(arg2, element) if i.positive? }
```

Comment:
- [ ] Your my_inject method is still not working with ranges. 
To understand why test your code with a range like (5..10) and see what happens.
Take also a moment to test in the irb console what methods that arrays have, you are being used by your code, and ranges don't have?

Reply:
This article is a bit lengthy but it has some interesting comparisons between arrays and ranges and might help your solution and understanding of ruby: [https://www.codesdope.com/ruby-array/](https://www.codesdope.com/ruby-array/)